example dev setup

michaelbest55 opened the pull request:
So here is what was suggested here: https://github.com/PyLabRobot/pylabrobot/pull/6#issuecomment-1264691856

I actually think a better way of doing this is to add `"sys_platform == "win32"'` to the install_requires in the pyhamilton library instead. Let me know what you think

rickwierenga commented:
PyHamilton already has `pythonnet` and `pywin32` in install_requires. What are you proposing to add?

Since we do not have tests for PyHamilton in PyLabRobot (yet?), I don't think it should be required in the dev target, Windows or not.

Also, please add the new target to CONTRIBUTING.md. I think maybe it should replace `all` in the 'from source' section.

Happy to discuss.

michaelbest55 commented:
I've removed the venus extras from the dev target. I've also replaced the mention of the all target in the installation docs for dev work.

Regarding pyhamilton, I'm suggesting the following https://github.com/dgretton/pyhamilton/blob/f23d4dd4fc4010db0e80bc56cfee44b6ae8f7119/setup.py#L18
should become
```
    install_requires=['requests', 'pythonnet', 'pywin32; sys_platform == "win32"'', 'pyserial']
```
This allows you to specify that you only want to install pywin if you are on a windows platform (which I would argue is what we would want).

rickwierenga commented:
Thanks! This looks great.

As for PyHamilton, since it only works on Windows, we don't care (in fact we prefer it) if it has Windows-only requirements.
